### ISSUE TITLE
NTR: Sleeps in 

### ISSUE THREAD

**Issue by arw36** [oborel/obo-relations#453]
I was looking for the most relevant relation for a primate using a specific tree species as a sleep sites. I couldn't find more generic "sleep", "rest" or "roost" relations between a animal and a plant. The two most closely related current terms are "creates habitat for" and "visits". Does "creates habitat for" have an inverse relation? Maybe visits could have subset of visit use such as feeding vs. resting? Let me know if a new term getting to specific and I should utilize a more generic term. I do think animal-habitat associations for sleeping would be widely applicable.

**Comment by cmungall** [comment]
any ideas @diatomsRcool @jhpoelen?

Planning for the future, are we likely to need a relation for every term in the behavior ontology, or can we come up with a good generic relation as @arw36 suggests

**Comment by arw36** [comment]
Thanks for these comments. To clarify, if "sleeps in" is too specific a term, I could see utilizing the inverse of "creates habitat for" as a more general term for this animal-plant relationship. However, as of yet, I don't see an inverse term for interaction. 

Visits could also suffice for this interaction, as I don't see a definition clarifying taxa or length of time focusing on certain animal-plant interactions. 

If a new term is warranted, I think "sleeps in", or whichever term is decided on, would be generalizable across lots of taxa. Te case study I describe is primates sleeping on branches of a live tree, similar to birds sleeping in nests in a tree. Other primates do construct nests, both up a tree (chimpanzees) and on the ground (gorillas).

**Comment by nlharris** [comment]
I really enjoy sleeping in, so I endorse this term request. 😴

**Comment by nlharris** [comment]
(That being said, has it been addressed by https://github.com/biolink/biolink-model/issues/850 or is there still a new term needed?)

**Comment by nlharris** [comment]
@diatomsRcool said "Yes, this can be closed" in biolink/biolink-model#850. Since that issue was already closed, I'm assuming she's referring to this one.

**Comment by diatomsRcool** [comment]
No, sorry I was confused. I was talking about the other one.

**Comment by nlharris** [comment]
So does this ticket need to be reopened?

**Comment by diatomsRcool** [comment]
I think so

**Comment by arw36** [comment]
Any updates on this? What do you need from me for this to be resolved?

**Comment by nlharris** [comment]
Hi @arw36, sorry this is taking so long. RO is a volunteer effort, and we currently meet only once every 8 weeks. I have added this request to the agenda for our next meeting (March 8).

**Comment by arw36** [comment]
The current definition for "creates habitat for" is: "An interaction relationship wherein one organism creates a structure or environment that is lived in by another organism." I propose that we create inverse for this "uses for habitat" or something. Then perhaps we include subset such as "is modified for habitat by" (and the inverse "modifies for habitat") in the case of a of plant material being used to make a nest.

**Comment by ddooley** [comment]
Thinking of a bird putting a nest together.  Some just fetch twigs, but also mud?  And various animals dig through dirt etc. to make tunnels. So a top-level habitat making model from creature's perspective, symbiotic or otherwise, should allow interventions that apply to inanimate materials as well as organisms.

**I'm wondering now if every behavioural object relation is a shortcut connecting between behavioural process model components?**

A "habitat creation process" has:

-    output some habitat
-    has participant some organism (as homebuilder)
-    occurs in some [host organism or place]
-    has input some inorganic material and/or organism resource

A model should provide for the distinction between inorganic and organism material, and may reference the location the behaviour process occurs in.

The "creates habitat for" relation sounds a bit intentional (ok if it is an evolved symbiotic relationship.) "establishes habitat for" might be more on-target: "An interaction relationship wherein a structure or environment established by one organism can be lived in by another organism.".  But this invites the inverse relations, and the need for material/nonmaterial reference:

- organism utilizes for habitat creation some material entity
- organism utilizes for habitat creation some inorganic material 
- organism utilizes for habitat creation some organism material 
- organism utilizes for habitat **location** some organism ...

Each of these maps from one part of the habitat creation process to another. And it looks like one gets into the categories of habitat location, for sleeping, eating, surveying, etc. Throw in complexity of obtaining building materials etc.

As an aside, in OBI process modelling, the first organism is both the "device" and the process participant?

**Comment by nlharris** [comment]
Are there still action items here to be addressed, or can this be closed now?

**Comment by jhpoelen** [comment]
@arw36 @diatomsRcool is the term "sleeps in" still to be added for your research?

**Comment by diatomsRcool** [comment]
No. That project didn't get funding. This can be closed from my perspective.

**Comment by jhpoelen** [comment]
@diatomsRcool bummer to hear that the project didn't get funded ; (

@arw36 would you still be interested in having "sleeps in" added to the Relations Ontology ?

**Comment by wdduncan** [comment]
I've added this topic to the [agenda](https://docs.google.com/document/d/19dhzj5QoliQZQgDJRczI878rLtexzMDo1g0K9_LRUWk/edit) for the upcoming Nov. 1 meeting.   
Let me know if you do not wish to discuss.

**Comment by wdduncan** [comment]
@cmungall @diatomsRcool will add a "has roost" under "has habitat" relation.

@arw36 Will this work for you?

**Comment by arw36** [comment]
Hi, can someone provide the status of this? It looks like [roosts in](http://purl.obolibrary.org/obo/RO_0008509) was added in 2023 -- awesome! 

Is there a complementary inverse to this? e.g., Banyan tree "is roost of" flying fox?

**Comment by anitacaron** [comment]
@arw36, can you create a PR to add this relation? If you need help, you can tag me 😄
